database/engine/rrdengineapi.c

```diff
@@ -54,6 +54,9 @@ void rrdeng_metric_init(RRDDIM *rd, uuid_t *dim_uuid)
     int replace_instead_of_generate = 0;
 
     ctx = rd->rrdset->rrdhost->rrdeng_ctx;
+    if (ctx == NULL && rd->rrdset->rrdhost != localhost)
+        ctx = localhost->rrdeng_ctx;
```

Comment:
I think these 2 lines should be a function because we use them everywhere and it's not clear what they mean.

Reply:
I will check, I think I already have something close to that